### ISSUE TITLE
Problem with bruker spx v1 loading

### ISSUE THREAD

**Issue by sem-geologist** [usnistgov/NeXLSpectrum.jl#3]
I want to point that there is a problem with loading spx files generated with Esprit v1, as often WorkingDistance tag is not saved to spx.
In case there is no WorkingDistance tag the code tries to multiply float (0.1) with Nothing which throws an error. Or maybe it is trying to stuff nothing into float64 and that throws error.
Error on my system:
```julia
julia> spec2 = loadspectrum("chevkinite-1.spx")
ERROR: type Nothing has no field content
Stacktrace:
 [1] getproperty(x::Nothing, f::Symbol)
   @ Base ./Base.jl:33
 [2] readbrukerspx(io::IOStream)
   @ NeXLSpectrum ~/.julia/packages/NeXLSpectrum/mEjAu/src/brukerspx.jl:154
 [3] loadspectrum(#unused#::Type{BrukerSPX}, ios::IOStream)
   @ NeXLSpectrum ~/.julia/packages/NeXLSpectrum/mEjAu/src/fileiosupport.jl:40
 [4] WARNING: both NeXLMatrixCorrection and NeXLCore export "plot2"; uses of it in module NeXLSpectrum must be qualified
(::NeXLSpectrum.var"#282#284"{Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}, DataType, String, NeXLSpectrum.var"#badname#283"})(ios::IOStream)
   @ NeXLSpectrum ~/.julia/packages/NeXLSpectrum/mEjAu/src/fileiosupport.jl:16
 [5] open(f::NeXLSpectrum.var"#282#284"{Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}, DataType, String, NeXLSpectrum.var"#badname#283"}, args::String; kwargs::Base.Iterators.Pairs{Symbol, Bool, Tuple{Symbol}, NamedTuple{(:read,), Tuple{Bool}}})
   @ Base ./io.jl:330
 [6] loadspectrum(ty::Type{BrukerSPX}, filename::String; kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ NeXLSpectrum ~/.julia/packages/NeXLSpectrum/mEjAu/src/fileiosupport.jl:15
 [7] loadspectrum(ty::Type{BrukerSPX}, filename::String)
   @ NeXLSpectrum ~/.julia/packages/NeXLSpectrum/mEjAu/src/fileiosupport.jl:10
 [8] loadspectrum(filename::String)
   @ NeXLSpectrum ~/.julia/packages/NeXLSpectrum/mEjAu/src/fileiosupport.jl:117
 [9] top-level scope
   @ REPL[6]:1

```

Reference to the code:
https://github.com/usnistgov/NeXLSpectrum.jl/blob/3fdbe61829681d31b7d60f3c4919c55f27298c8b/src/brukerspx.jl#L154-L155

proposition to fix:
replace those lines with:
```julia
        wdxmlnode = findfirst("WorkingDistance", item)  
        if !isnothing(wdxmlnode)                                
            props[:WorkingDistance] = 0.1 * parse(Float64, wdxmlnode.content)                 
        end
```

**Comment by sem-geologist** [comment]
oh, I missed the precise error description
`ERROR: type Nothing has no field content`

being python dev I had not understood what this `field` is (we call these `attributes`). So it fails earlier than trying to cast it to `Float64`

**Comment by NicholasWMRitchie** [comment]
I've fixed the problem in the "develop" branch.  You are right it occurs a little earlier when a `nothing` is passed to the parse function.

**Comment by NicholasWMRitchie** [comment]
I forgot to mention, please tell me if this fix works for you.  I don't have a sufficiently old Bruker spectrum to test it on.
Even better would be to send me a spectrum that fails.

**Comment by sem-geologist** [comment]
Proposed addition fixed issue for me when I tested that. I am attaching the file below which was causing trouble.
Below is the link to the spx as github gist (You will need to click "raw" and save as file with spx extension).
https://gist.github.com/sem-geologist/ffb39911ac116a762af8a1794652a2ab

**Comment by NicholasWMRitchie** [comment]
Resolved in 00994aa and 1eabf76
